upgrade/v17xto180/upgrade.go

```diff
@@ -227,6 +232,33 @@ func upgradeBackingImages(namespace string, lhClient *lhclientset.Clientset, res
 	return nil
 }
 
+func upgradeSystemBackups(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, upgradeLogPrefix+"upgrade system backup failed")
+	}()
+
+	sbMap, err := upgradeutil.ListAndUpdateSystemBackupsInProvidedCache(namespace, lhClient, resourceMaps)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to list all existing Longhorn system backups during the system backup upgrade")
+	}
+
+	defaultBackupTarget, err := lhClient.LonghornV1beta2().BackupTargets(namespace).Get(context.TODO(), types.DefaultBackupTargetName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get default backup target")
+	}
+
+	for _, sb := range sbMap {
+		if sb.OwnerReferences == nil {
+			sb.OwnerReferences = datastore.GetOwnerReferencesForBackupTarget(defaultBackupTarget)
+		}
+	}
+
+	return nil
+}
+
```

Comment:
Have you tested the upgrade path and check the owner reference are patched correctly?

Reply:
Yes, tested.